project/plugins.sbt

```diff
@@ -1,4 +1,2 @@
-// Comment to get more information during initialization
-logLevel := Level.Info
-
-addSbtPlugin("org.foundweekends" % "sbt-bintray" % "0.5.6")
\ No newline at end of file
+addSbtPlugin("com.github.sideeffffect" % "sbt-decent-scala" % "0.4.0+7-6526cb25")
```

Comment:
Wouldn't it be better to use the plugins directly?

Reply:
it was just much more comfortable to do it this way, instead of writing it all manually.
here's the source code of the plugin, if you would like to have a look https://github.com/sideeffffect/sbt-decent-scala

Reply:
I checked the code before I wrote the comment, of course 😄 I belive such _aggregating plugin_ is valuable inside a company (or very large open-source project) but here I'm afraid that it would make the project less accessible for contributors because it introduces a next level of _abstraction_.

If @jendakol is OK with this, I will not block its usage.

Reply:
Will you please adress also this?

Reply:
I'm not a fan of this.
1) What is that plugin? It would be better to use somewhat "approved" plugins in company projects.
1) The plugin is used to aggregate some _useful_ plugins, however, it does also hide them at the same moment - so for a newcomer, it's not visible that some plugins are used (like e.g. `better-monadic-for`).
1) What's also a problem for me is that specifically `better-monadic-for` is used here - we _banned_ its usage in projects in internal GH and now you propose (well, it's not even visible) to use it in a public project? :neutral_face: The same situation is with the SBT wrapper BTW - it was _banned_ and here I see it used. Why is it used, why is SBT itself not enough? And for hats is `better-monadic-for` used here in the specific project?

Reply:
OK, I'll inline the config

Reply:
So I've inlined most of `sbt-decent-scala`, it was some work, but I understand that using a "private" plugin on a company project may not be the best thing (or may need a more deliberation). Some things were left out, since there's no immediate need for them (but we can add them once such need arises, of course), namely: silencer, bettern-monadic-for and kind-projector.

Regarding the sbt script, its benefit is that it makes the repository self-standing/self-contained (almost, the only thing we then depend on is bash and JVM) and is fully version in the git repo with the code. That means that we have as few demands/dependencies on the CI environment as possible and thus can be sure the CI will behave always the same. We have Travis, now also GitHub Actions and @augi also floated the idea of using GH Actions with bare JVM, so I hope you see how that's relevant. And the fact that it's versioned in git ensures that the git snapshot determines the CI behavior to the maximum extent.
(And it also makes the project compilable to people without sbt installed, which is common, or people without IDE, which is much less common, but still. But this is a minor point, of course.)

This is just a side note, since here is probably not the best forum to discuss this, but I have trouble remembering when we have banned either better-monadic-for or the script.

Reply:
Please note that I like the idea of `sbt wrapper`. Sadly, it is not an official wrapper and Windows version is missing, but it is beneficial IMHO.

Reply:
My problem is that SBT wrapper is an unofficial thing and it even doesn't fit 1:1 to the SBT itself. I believe there was a huge discussion in internal GH where @jakubjanecek explained his objections and while my opinion is maybe not as strong as his, in general, I agree with this is something we probably want to avoid if possible.